File: scripts/btc/lock/rpc_call.go
/*

 */

package main

import (
	"bytes"
	"encoding/json"
	"errors"
	"fmt"
	"io/ioutil"
	"net/http"
)

type jsonRPCData struct {
	Method  string                 `json:"method"`
	Params  map[string]interface{} `json:"params"`
	ID      int                    `json:"id"`
	JsonRpc string                 `json:"jsonrpc"`
}
type RPCResponse struct {
	Result map[string]interface{} `json:"result"`
	Error  map[string]interface{} `json:"error"`
}

func makeRPCcall(method string, params map[string]interface{}) (*RPCResponse, error) {

	url := "http://127.0.0.1:26602/jsonrpc"

	payload, _ := json.Marshal(&jsonRPCData{
		Method:  method,
		Params:  params,
		ID:      51,
		JsonRpc: "2.0",
	})

	req, _ := http.NewRequest("POST", url,
		bytes.NewReader(payload))

	req.Header.Add("Content-Type", "application/json")
	req.Header.Add("Accept", "application/json")

	res, err := http.DefaultClient.Do(req)
	if err != nil {
		fmt.Println("json rpc error", err)
		return nil, err
	}

	defer res.Body.Close()
	body, _ := ioutil.ReadAll(res.Body)

	if res.StatusCode != 200 {
		panic(string(body))
	}

	fmt.Println(string(body))
	resp := RPCResponse{}
	err = json.Unmarshal(body, &resp)
	if err != nil {
		return nil, err
	}

	if resp.Error != nil {
		fmt.Println(resp.Error)
		return nil, errors.New("rpc error")
	}

	return &resp, nil

}

File: scripts/ons/sdk/rpc_call.py
import requests
import json

#url = "http://127.0.0.1:26600/jsonrpc"

url = "http://35.184.133.167:26606/jsonrpc"    #sdk url
faucet_url = "http://34.66.255.58:26607/jsonrpc"

headers = {
    "Content-Type": "application/json",
    "Accept": "application/json",
}

def rpc_call(method, params):
    payload = {
        "method": method,
        "params": params,
        "id": 123,
        "jsonrpc": "2.0"
    }

    response = requests.request("POST", url, data=json.dumps(payload), headers=headers)

    if response.status_code != 200:
        return ""

    resp = json.loads(response.text)
    return resp


def faucet_call(method, params):
    payload = {
        "method": method,
        "params": params,
        "id": 123,
        "jsonrpc": "2.0"
    }

    response = requests.request("POST", faucet_url, data=json.dumps(payload), headers=headers)

    if response.status_code != 200:
        return ""

    resp = json.loads(response.text)
    return resp




def converBigInt(value):
    return str(value)
